tico/utils/padding.py

```diff
@@ -12,36 +12,71 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from enum import IntEnum
+from typing import NamedTuple, Optional, Sequence, Tuple, Union
+
 import torch
 
 from tico.utils.errors import InvalidArgumentError
 
-SAME = 0
-VALID = 1
 
+PaddingValue = Union[str, Sequence[int]]  # "same" | "valid" | [pad_h, pad_w]
 
-def is_valid_padding(padding: str | list):
-    if isinstance(padding, str):
-        return padding == "valid"
 
-    if isinstance(padding, list):
-        assert len(padding) == 2, "Padding should be a list of length 2."
-        return padding == [0, 0]
+class ConvPadding(IntEnum):
+    SAME = 0  # auto-pad, HW out == HW in
+    VALID = 1  # no implicit padding
+
 
-    raise InvalidArgumentError("Invalid padding.")
+class ConvPaddingInfo(NamedTuple):
+    """
+    Result of padding analysis.
+    """
 
+    conv_padding_type: ConvPadding
+    explicit_pad_hw: Optional[Tuple[int, int]]  # None -> no extra Pad() op needed
 
-def is_same_padding(
-    padding: str | list, input_shape: list | torch.Size, output_shape: list | torch.Size
-):
+
+def identify_padding(
```

Comment:
For stride != [1,1], normal conv and transposed conv have different logics to identify the same padding.